src/sinks/pulsar/config.rs

```diff
@@ -0,0 +1,292 @@
+
+    /// The OAuth2 audience.
+    #[configurable(metadata(docs::examples = "${OAUTH2_AUDIENCE}"))]
+    #[configurable(metadata(docs::examples = "pulsar"))]
+    audience: Option<String>,
+
+    /// The OAuth2 scope.
+    #[configurable(metadata(docs::examples = "${OAUTH2_SCOPE}"))]
+    #[configurable(metadata(docs::examples = "admin"))]
+    scope: Option<String>,
+}
+
+/// Supported compression types for Pulsar.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Derivative)]
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+pub enum PulsarCompression {
+    /// No compression.
+    #[derivative(Default)]
+    None,
+
+    /// LZ4.
+    Lz4,
+
+    /// Zlib.
+    Zlib,
+
+    /// Zstandard.
+    Zstd,
+
+    /// Snappy.
+    Snappy,
+}
+
+impl Default for PulsarSinkConfig {
+    fn default() -> Self {
+        Self {
+            endpoint: "pulsar://127.0.0.1:6650".to_string(),
+            topic: "topic-1234".to_string(),
+            producer_name: None,
+            properties_key: None,
+            partition_key_field: None,
+            batch: Default::default(),
+            compression: Default::default(),
+            encoding: TextSerializerConfig::default().into(),
+            auth: None,
+            acknowledgements: Default::default(),
+        }
+    }
+}
+
+impl PulsarSinkConfig {
+    pub(crate) async fn create_pulsar_client(&self) -> Result<Pulsar<TokioExecutor>, PulsarError> {
+        let mut builder = Pulsar::builder(&self.endpoint, TokioExecutor);
+        if let Some(auth) = &self.auth {
+            builder = match (
+                auth.name.as_ref(),
+                auth.token.as_ref(),
+                auth.oauth2.as_ref(),
+            ) {
+                (Some(name), Some(token), None) => builder.with_auth(Authentication {
+                    name: name.clone(),
+                    data: token.inner().as_bytes().to_vec(),
+                }),
+                (None, None, Some(oauth2)) => builder.with_auth_provider(
+                    OAuth2Authentication::client_credentials(OAuth2Params {
+                        issuer_url: oauth2.issuer_url.clone(),
+                        credentials_url: oauth2.credentials_url.clone(),
+                        audience: oauth2.audience.clone(),
+                        scope: oauth2.scope.clone(),
+                    }),
+                ),
+                _ => return Err(PulsarError::Authentication(AuthenticationError::Custom(
+                    "Invalid auth config: can only specify name and token or oauth2 configuration"
+                        .to_string(),
+                ))),
+            };
+        }
+
+        builder.build().await
+    }
+
+    pub(crate) fn build_producer_options(&self) -> ProducerOptions {
+        let mut opts = ProducerOptions {
+            encrypted: None,
+            access_mode: Some(0),
+            metadata: Default::default(),
+            schema: None,
+            batch_size: None,
```

Comment:
```suggestion
            batch_size: Some(self.batch.max_events),
```

Why not do the batch size in one line?